src/main/util/helper_classes/id_counter.py

```diff
@@ -1,15 +1,25 @@
 # Copyright (c) 2020 Anastasiia Birillo, Elena Lyulina
 
-from typing import Dict
+import logging
+from typing import Dict, Any, Type
 from collections import defaultdict
 
+from src.main.util.consts import LOGGER_NAME
+from src.main.util.log_util import log_and_raise_error
+from src.main.util.default_dict_util import get_default_dict_with_default_dict_with_none, get_none
+
+log = logging.getLogger(LOGGER_NAME)
+
 
 class IdCounter:
     _instances: Dict[str, int] = defaultdict(int)
+    _id_item_dict_by_class: Dict[str, Dict[int, Any]] = defaultdict(get_default_dict_with_default_dict_with_none)
```

Comment:
I think since we store items in this dict, their type is not Any, but Type[IdCounter], what do you think?